client/src/components/Map.tsx

```diff
@@ -79,19 +80,33 @@ export default class Map extends Component<Prop>{
             .attr("stroke-width", 150)
 
             if(index > 0){
+                
+                svg.append("marker")
+                .attr("id", "triangle")
+                .attr("refX", -10)
+                .attr("refY", 6)
+                .attr("markerWidth", 30)
+                .attr("markerHeight", 30)
+                .attr("markerUnits","userSpaceOnUse")
+                .attr("orient", "auto")
```

Comment:
Here, I am reviewing the added direction arrows feature. You found a great solution and implemented it very cleanly.

I too think that the "auto" orientation is the problem. I have a solution suggestion:
- replace auto with a number - degrees
- e.g. `.attr("orient", 90)`. Try it out. ([the docs](https://www.w3.org/TR/svg-markers/#OrientAttribute))
- you can calculate the angle because you have the two coordinates (xy1 and xy2)
- I suggest implementing the "find angle" function in `src/utils/Math.ts` file.
- You can use math (trigonometry) for that, but I suggest generating the function with ChatGPT and implementing it manually

I want to point out that the coordinates are relative in percentage. That would make your calculated angle a bit off because the width and the height aren't the same. The fix for that is multiplying relative coordinates with the submap width/heigh that you have in the prop.

Wow! That was a longer explanation than I anticipated. But that solution should guarantee always correct arrows.

Great job on finding the bug, in the first place.